src/test/java/org/broadinstitute/consent/http/resources/DaaResourceTest.java

```diff
@@ -164,9 +169,110 @@ void testFindDaaByDaaId() {
+    admin.setRoles(List.of(role));
+    admin.setInstitutionId(1);
+    DataAccessAgreement daa = new DataAccessAgreement();
+    daa.setDaaId(1);
+    LibraryCard lc = new LibraryCard();
+    lc.setInstitutionId(1);
+
+    when(daaService.findById(any())).thenReturn(daa);
+    when(userService.findUserByEmail(any())).thenReturn(admin);
+    when(libraryCardService.findLibraryCardById(any())).thenReturn(lc);
+
+    resource = new DaaResource(daaService, dacService, userService, libraryCardService);
+    Response response = resource.createLibraryCardDaaRelation(info, authUser, daa.getDaaId(),  lc.getId());
+    assert response.getStatus() == HttpStatus.SC_CREATED;
+  }
+
+  @Test
+  void testCreateLibraryCardDaaRelation_SigningOfficialCase() {
+    UriInfo info = mock(UriInfo.class);
+    UriBuilder builder = mock(UriBuilder.class);
+    when(info.getBaseUriBuilder()).thenReturn(builder);
+    when(builder.replacePath(any())).thenReturn(builder);
+    Dac dac = new Dac();
+    dac.setDacId(RandomUtils.nextInt(10, 100));
+    User admin = new User();
+    UserRole role = (new UserRole(UserRoles.SIGNINGOFFICIAL.getRoleId(), UserRoles.SIGNINGOFFICIAL.getRoleName()));
+    admin.setRoles(List.of(role));
+    admin.setInstitutionId(1);
+    DataAccessAgreement daa = new DataAccessAgreement();
+    daa.setDaaId(1);
+    LibraryCard lc = new LibraryCard();
+    lc.setInstitutionId(1);
+
+    when(daaService.findById(any())).thenReturn(daa);
+    when(userService.findUserByEmail(any())).thenReturn(admin);
+    when(libraryCardService.findLibraryCardById(any())).thenReturn(lc);
+
+    resource = new DaaResource(daaService, dacService, userService, libraryCardService);
+    Response response = resource.createLibraryCardDaaRelation(info, authUser, daa.getDaaId(),  lc.getId());
+    assert response.getStatus() == HttpStatus.SC_CREATED;
+  }
+
+  @Test
+  void testCreateLibraryCardDaaRelation_InvalidInstitutionIdCase() {
+    UriInfo info = mock(UriInfo.class);
+    UriBuilder builder = mock(UriBuilder.class);
+    Dac dac = new Dac();
+    dac.setDacId(RandomUtils.nextInt(10, 100));
+    User admin = new User();
+    UserRole role = (new UserRole(UserRoles.SIGNINGOFFICIAL.getRoleId(), UserRoles.SIGNINGOFFICIAL.getRoleName()));
+    admin.setRoles(List.of(role));
+    admin.setInstitutionId(2);
+    DataAccessAgreement daa = new DataAccessAgreement();
+    daa.setDaaId(1);
+    LibraryCard lc = new LibraryCard();
+    lc.setInstitutionId(1);
+
+    when(daaService.findById(any())).thenReturn(daa);
+    when(userService.findUserByEmail(any())).thenReturn(admin);
+    when(libraryCardService.findLibraryCardById(any())).thenReturn(lc);
+
+
+    resource = new DaaResource(daaService, dacService, userService, libraryCardService);
+    Response response = resource.createLibraryCardDaaRelation(info, authUser, daa.getDaaId(),  lc.getId());
+    assert response.getStatus() == HttpStatus.SC_FORBIDDEN;
+  }
+
+  @Test
```

Comment:
Full coverage of all the lines would be beneficial, for example with an invalid or null user and perhaps other test cases.